### PR TITLE
Add a script to cull client IP entries

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

File: synapse/storage/client_ips.py [hunk 1]
```diff
@@ -223,7 +223,12 @@ def _get_last_client_ip_by_device_txn(cls, txn, user_id, device_id, retcols):
 
     @defer.inlineCallbacks
     def get_user_ip_and_agents(self, user):
-        user_id = user.to_string()
+
+        if isinstance(user, str):
```
Comment (richvdh) [inline synapse/storage/client_ips.py]: can you add a docstring which documents the acceptable types of `user`?

Comment (richvdh) [inline synapse/storage/client_ips.py]: Why is this necessary, btw? is there not a danger of `user` being a `unicode`?

File: tests/utils.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ class TestHomeServer(HomeServer):
 
 @defer.inlineCallbacks
 def setup_test_homeserver(
-    cleanup_func,
+    self,
```
Comment (richvdh) [inline tests/utils.py]: nit: could this be called test_object or something, since it's not a real instance method?

File: tests/utils.py [hunk 3]
```diff
@@ -182,8 +182,7 @@ def setup_test_homeserver(
     If no datastore is supplied, one is created and given to the homeserver.
 
     Args:
-        cleanup_func : The function used to register a cleanup routine for
-                       after the test.
+        self: The UnitTest class.
```
Comment (richvdh) [inline tests/utils.py]: I think it needs to be a TestCase (instance) rather than a UnitTest (class)?

```suggestion
        self (twisted.trial.unittest.TestCase): The test instance
```

**Comment by hawkowl** [comment]
Don't need this for the moment, so I'll close it.
